Data.Rmd

```diff
@@ -0,0 +1,329 @@
+
+The snippet above is something the maintainer executes once (or every time they need to update `my_pkg_data`).
+This is workflow code and should **not** appear in the `R/` directory of the source package.
+(We'll talk about a suitable place to keep this code below.) For larger datasets, you may want to experiment with the compression setting, which is under the control of the `compress` argument.
+The default is "bzip2", but sometimes "gzip" or "xz" can create smaller files.
+
+It's possible to use other types of files below `data/`, but we don't recommend it because `.rda` files are already fast, small, and explicit.
+The other possibilities are described in the documentation for `utils::data()` and in the [Data in packages](https://rstudio.github.io/r-manuals/r-exts/Creating-R-packages.html#data-in-packages) section of Writing R Extensions.
+In terms of advice to package authors, the help topic for `data()` seems to implicitly make the same recommendations as we do above:
+
+-   Store one R object in each `data/*.rda` file
+-   Use the same name for that object and its `.rda` file
+-   Use lazy-loading, by default
+
+If the `DESCRIPTION` contains `LazyData: true`, then datasets will be lazily loaded.
+This means that they won't occupy any memory until you use them.
+The following example shows memory usage before and after loading the nycflights13 package.
+You can see that memory usage doesn't change until you inspect the `flights` dataset stored inside the package.
+
+```{r}
+pryr::mem_used()
+library(nycflights13)
+pryr::mem_used()
+
+invisible(flights)
+pryr::mem_used()
+```
+
+We recommend that you include `LazyData: true` in your `DESCRIPTION` if you are shipping `.rda` files below `data/`.
+If you use `use_data()` to create such datasets, it will automatically make this modification to `DESCRIPTION` for you.
+
+In terms of advice to package users, note that lazily-loaded datasets do **not** need to be pre-loaded.
+Above, once we did `library(nycflights13)`, we could immediately access `flights`.
+There is no call `data(flights)`, because it is not necessary.
+
+There are specific downsides to `data(some_pkg_data)` calls that support a policy of only using `data()` when it is actually necessary, i.e. for datasets that would not be available otherwise:
+
+-   By default `data()`, creates objects in the user's global workspace. There is the potential to silently overwrite pre-existing objects with new values.
+-   There is also no guarantee that `data(foo)` will create exactly one object named "foo". It could create more than one object and/or objects with totally different names.
+
+### Preserve the origin story of package data {#sec-data-data-raw}
+
+Often, the data you include in `data/` is a cleaned up version of raw data you've gathered from elsewhere.
+We highly recommend taking the time to include the code used to do this in the source version of your package.
```

Comment:
If you want a counter-example: we have no way to recreate the ggplot2 datasets. They're from scripts that existed on Hadley's computer at some time in the past, but not any more.